manifests/database/default_read_grant.pp

```diff
@@ -19,7 +19,7 @@
                   acl.defaclacl
                 FROM pg_default_acl acl
                 JOIN pg_namespace ns ON acl.defaclnamespace=ns.oid
-                WHERE acl.defaclacl::text ~ '.*\\\\\"${database_read_only_username}\\\\\"=r/${database_username}\\\".*'
+                WHERE acl.defaclacl::text ~ '.*(\\\\\")?${database_read_only_username}(\\\\\")?=r/(\\\\\")${database_username}(\\\\\")?.*'
```

Comment:
Did you mean:

```suggestion
                WHERE acl.defaclacl::text ~ '.*(\\\\\")?${database_read_only_username}(\\\\\")?=r/(\\\\\")?${database_username}(\\\\\")?.*'
```

In seems to be missing only on this query.

Reply:
Also, I think you are expecting balanced quotes, one of:
1. `foo=r/bar`
2. `foo=r/"b-a-r"`
3. `"f-o-o"=r/bar`
4. `"f-o-o"=r/"b-a-r"`

In this case you can use back-reference to previous captures:
```suggestion
                WHERE acl.defaclacl::text ~ '.*((?:\\\\\")?)${database_read_only_username}\\1=r/((?:\\\\\")?)${database_username}\\2.*'
```

The leading and trailing `.*` can also be removed, but it might make more sense to match the full field:

```suggestion
                WHERE acl.defaclacl::text ~ '^{"((?:\\\\\")?)${database_read_only_username}\\1=r/((?:\\\\\")?)${database_username}\\2"}$'
```

The same apply for the other 2 queries.

Reply:
Thank you @smortex for corrections. I'm totally agreed with the first one but two others I have to test. I'm not a great expert in regex.

Reply:
For me, there were some additional `\` needed due to the escaping hell from puppet to bash to psql.

This worked for me:
```
                WHERE acl.defaclacl::text ~ '^{\"((?:\\\\\\\\\")?)${database_read_only_username}\\1=r\\/((?:\\\\\\\\\")?)${database_username}\\2\"}$'
                WHERE acl.defaclacl::text ~ '^{\"((?:\\\\\\\\\")?)${database_read_only_username}\\1=U\\/((?:\\\\\\\\\")?)${database_username}\\2\"}$'
                WHERE acl.defaclacl::text ~ '^{\"((?:\\\\\\\\\")?)${database_read_only_username}\\1=X\\/((?:\\\\\\\\\")?)${database_username}\\2\"}$'
```